### PR TITLE
Deploy UI code to Azure (Storage) when PR is complete (GH Actions)

### LINKED ISSUE #21: [Task] Deploy UI code to Azure (Storage) when PR is complete (GH Actions)
**Is your feature request related to a problem? Please describe.**
The goal of this feature is to implement GitHub Actions that allow us to flexibly build an automated software development lifecycle workflow within GitHub in order to speed up the development process.

**Describe the solution you'd like**
The solution I found in the documentation, `https://docs.microsoft.com/en-us/azure/developer/github/github-actions`
describes a solution called a _workflow_, which is an automated process that you set up in your GitHub repository and is made up of individual actions that run after a specific event (like a pull request) occur. The individual actions are packaged scripts that automate software development tasks.

**Describe alternatives you've considered**
I believe the same thing can be accomplished with Azure Pipelines.

### PR DESCRIPTION
**Pull request by andluc21**
Closes #21  - Team 3 Task 6

This feature adds a GitHub Actions workflow that automatically deploys a static website to the Azure storage account container from GitHub once you make changes to the site's code, after the pull request is complete. 

- The _.github/workflows/AzureDeploy.yml_ file defines the workflow configuration

1. In order to test this feature first, create a new branch that you can merge the `development` branch into while still having two separate branches. You will be using the newly created branch to push changes/ make pull requests from,  in order to update the site's code here.  https://nordstromui.z5.web.core.windows.net/ 
2.   To update the site's code set up a `index.html` file and a `404.html` file  in the root folder of the repository within your separate branch.
-_ad440-winter2021-thursday-repo/index.html_  
-_ad440-winter2021-thursday-repo/404.html_  

3. Make updates to the `index.html` file.
4. Once you push your changes to GitHub, make a pull request from your branch to the `development` branch.
5. Here we will see the Github Action workflow first pop up and  make a request for a check to be passed before making the merge request.
6. After making the Pull Request, Go to Actions for your GitHub repository. Open the first result to see detailed logs of the workflow's run.
7. Lastly the changes made, will be updated to the static website https://nordstromui.z5.web.core.windows.net/

**Troubleshooting Tips**

it is possible that you may need my specific secret for the input credentials of the Azure Login action, located in the AzureDeploy.yml file `creds: ${{ secrets.AZURE_CREDENTIALS }}` that grabs those credentials from  the stored secrets in GitHubs settings if it does not transfer, But to deploy this workflow to your own Azure storage account container by creating a your own resource group, storage container with static website enabled giving your own name as well. Then 
 creating a service principal with  Azure Cloud Shell/PowerShell, using this command
 ```
az ad sp create-for-rbac --name {myStaticSite} --role contributor --scopes /subscriptions/{subscription-id}/resourceGroups/{resource-group} --sdk-auth 
```
and pasting the entire JSON output from the Azure CLI command into the Secret's value field under GitHubs Settings.

_Source: https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-static-site-github-actions_

Date | Activity | Time Spent
-- | -- | --
1/11/2021|Research GH Actions documentation|2 hours
1/13/2021|Research hosting websites in Azure|2 hours
1/15/2021|Follow documentation to implement web app in Azure | 2 hours 
1/16/2021|Follow documentation that directly accomplishes the Task | 3 hours

### CODE REVIEW

File: .github/workflows/AzureDeploy.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Blob storage website CI
+
+on:
+  pull_request:
+    branches:
+      - development
+      - TestBranch
```
Comment (zxkevinxz) [inline .github/workflows/AzureDeploy.yml:7]: Using on.pull_request will trigger this action whenever a PR is created, and not when it is completed/merged. Should this be on.push instead?

**Review by zxkevinxz — DISMISSED** [review]
This is something that you, @lbrown51 and I should've discussed beforehand, but we all used different naming conventions for our yml file. We should probably decide on 1 style.

File: .github/workflows/AzureDeploy.yml (hunk 1 above)
Comment (lbrown51) [reply .github/workflows/AzureDeploy.yml:7]: Further question, should the test branch be included here?

Comment (andluc21) [reply .github/workflows/AzureDeploy.yml:7]: That was not the case when I tested this. It would Re-Deploy as soon as the PR was complete.

**Review by unicornsleuth — APPROVED** [review]
It seems to have worked! The code looks clean, and I see the Action ran, and it did update the site!

**Review by zxkevinxz — APPROVED** [review]
Agree with Larissa, and this is the 2nd time I've checked it out. Good work.

**Pull request merged**